guidance/guidance_los/scripts/los_guidance_action_server.py

```diff
@@ -0,0 +1,367 @@
+#!/usr/bin/env python3
+
+import os
+from dataclasses import dataclass
+
+import numpy as np
+import rclpy
+import yaml
+from geometry_msgs.msg import PoseStamped, Vector3Stamped
+from guidance_los.los_guidance_algorithm import ThirdOrderLOSGuidance
+from nav_msgs.msg import Odometry
+from rclpy.action import ActionServer
+from rclpy.action.server import ServerGoalHandle
+from rclpy.callback_groups import ReentrantCallbackGroup
+from rclpy.executors import MultiThreadedExecutor
+from rclpy.node import Node
+from std_msgs.msg import String
+from transforms3d.euler import quat2euler
+from vortex_msgs.action import NavigateWaypoints
+from vortex_msgs.msg import LOSGuidance
+
+
+@dataclass(slots=True)
+class State:
+    """Data class representing the current state of the vehicle."""
+
+    x: float = 0.0
+    y: float = 0.0
+    z: float = 0.0
+    yaw: float = 0.0
+    pitch: float = 0.0
+
+    def as_ndarray(self) -> np.ndarray:
+        """Convert state to numpy array for guidance calculations."""
+        return np.array([self.x, self.y, self.z, self.yaw, self.pitch])
+
+
+class GuidanceActionServer(Node):
+    """ROS2 Action Server node for waypoint navigation using LOS guidance."""
```

Comment:
I think the class should be called LOSActionServer, as it will not be the only Guidance action server.